### PR TITLE
Add object properties multiline padding rule

### PR DESCRIPTION
**Pull request by quesabe**
Closes PLA-78.

### CODE REVIEW

**Review by Serator — APPROVED** [review]

**Review by gvidon — CHANGES_REQUESTED** [review]
@quesabe please add a short doc in the `README.md`. You can follow other rules as example.

File: lib/rules/object-properties-multiline-padding.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+          const nextProperty = properties[i + 1]
+
+          // No padding at the end.
+          if(!nextProperty && isEmptyLine(objectEndLine - propertyEndLine)) {
+            context.report({messageId: 'redundant-after', node: property})
+          }
+
+          // Multi-property object
+          if(previousProperty) {
```
Comment (gvidon) [inline lib/rules/object-properties-multiline-padding.js:45]: @quesabe neat stuff 🤓 Is there a way we can avoid nested `if` statements?

File: tests/lib/rules/jsx-newline-block.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 'use strict'
 
 // ANCHOR Requirements
-const rule = require('../../../lib/rules/jsx-newline-block')
-const {RuleTester } = require('eslint')
+const ruleName = 'jsx-newline-block'
```
Comment (gvidon) [inline tests/lib/rules/jsx-newline-block.js:4]: @quesabe 👍

**Review by gvidon — CHANGES_REQUESTED** [review]

**Review by gvidon — CHANGES_REQUESTED** [review]
@quesabe do you think we can implement [auto fix](https://eslint.org/docs/developer-guide/working-with-rules#applying-fixes) for this rule?

File: lib/rules/object-properties-multiline-padding.js (hunk 1 above)
Comment (quesabe) [reply lib/rules/object-properties-multiline-padding.js:45]: Yes, it is possible. But I thought that one-level nesting is a little cleaner (see the code below). Let me know which option is preferred.

```typescript
// No nesting
if(
  previousProperty && 
  isMultiline(property) && 
  isNoEmptyLine(propertyStartLine - previousProperty.loc.end.line)
) {
  context.report({messageId: 'require-before', node: property})
}

if(
  previousProperty && 
  !isMultiline(property) && 
  !isMultiline(previousProperty) && 
  isEmptyLine(propertyStartLine - previousProperty.loc.end.line)
) {
  context.report({messageId: 'redundant-before', node: property})
}

// One level nesting
if(nextProperty) {
  const nextPropertyStartLine = nextProperty.loc.start.line
  const distanceToNext = nextPropertyStartLine - propertyEndLine

  if(isMultiline(property) && isNoEmptyLine(distanceToNext)) {
    context.report({messageId: 'require-after', node: property})
  }

  if(!isMultiline(property) && !isMultiline(nextProperty) && isEmptyLine(distanceToNext)) {
    context.report({messageId: 'redundant-after', node: property})
  }
}
```

Comment (gvidon) [reply lib/rules/object-properties-multiline-padding.js:45]: @quesabe yeah, actually I thought about it again. Let's keep it like this.

**Review by gvidon — APPROVED** [review]
@Serator do you have any comments?

File: lib/rules/object-properties-multiline-padding.js (hunk 1 above)
Comment (quesabe) [reply lib/rules/object-properties-multiline-padding.js]: Finally I did a simpler flow and removed a level of conditions. Now it looks even better :)

**Comment by quesabe** [comment]
> @quesabe do you think we can implement [auto fix](https://eslint.org/docs/developer-guide/working-with-rules#applying-fixes) for this rule?

Added the fix and covered it in tests. Please take a look.

**Review by gvidon — APPROVED** [review]

**Pull request merged**